### ISSUE TITLE
[Improvement][Discussion] Should PRE/POST update events be fired when autosave takes place

### ISSUE THREAD

**Issue by fashxp** [pimcore/pimcore#11474]
### Improvement description

might cause to unwanted behavior, like https://github.com/pimcore/advanced-object-search/issues/143 and the likes. 

question is, can auto-save versions be interesting to some extensions?

**Comment by NiklasBr** [comment]
I agree on the usefulness of not triggering events on autosave, but rather than not trigger them at all I would propose that these events also supply a context information similar to how `ElementAdminStyleEvent` does it:

https://github.com/pimcore/pimcore/blob/41ea483e2dc0cd0dd6017797c2a2252b32a28f78/lib/Event/Admin/ElementAdminStyleEvent.php#L24-L39

So... Update `ElementEventInterface` or `DataObjectEvents` to add a `getContext()` method which can be called in listeners with something like this:

```php
class ObjectListener implements EventSubscriberInterface
{
    public static function getSubscribedEvents(): array
    {
        return [
            DataObjectEvents::PRE_UPDATE => 'myPreUpdate',
        ];
    }

    public function myPreUpdate(ElementEventInterface $elementEvent)
    {
        if ($elementEvent->getContext() !== ElementEventInterface::CONTEXT_AUTOSAVE) {
            // ...
        }
    }
}
```

**Comment by fashxp** [comment]
we have already the information in arguments of the event ... so that is not the problem.

**Comment by NiklasBr** [comment]
> we have already the information in arguments of the event ... so that is not the problem.

Oh, I did not know that! Looked in the documentation and found nothing.

**Comment by passioneight** [comment]
Related: https://github.com/pimcore/pimcore/issues/11743
